src/ID.h

```diff
@@ -95,6 +94,11 @@ class ID final : public Obj, public notifier::detail::Modifiable {
 
     void ClearVal();
 
+    // *If* this identifier is a function parameter, return the record
+    // associated with its declaration, otherwise nil.
+    RecordTypePtr GetParamRecord() const { return param_record; }
+    void SetParamRecord(const RecordTypePtr& pr) { param_record = pr; }
```

Comment:
This is quirky, but I think I understand where it comes from (the now removed comment about renaming was useful, maybe that could be added back here?

After long staring:

>             if ( profiled_func_t && id->GetParamRecord() == profiled_func_t->Params() )

Is it too naive to do the following:
In `ProfileFunc::ProfileFunc`, go via `func->GetScope()->OrderedVars()` and take the first `func->GetType()-Params().NumFields()` from it. Shouldn't those be the IDs representing parameters? If `id` is in those, it's a function parameter.


If that's no dice, I think would vote for attaching the full `FuncPtr` to the ID rather than just the params. Just so that you could have more context if needed.

Reply:
Your suggestion does indeed work, likewise thanks! I was concerned it might run into problems due to inlining and/or lambdas, but I've tested those and it remains valid, and much simpler